src/main/java/com/dev/cinema/controller/OrderController.java

```diff
@@ -0,0 +1,45 @@
+package com.dev.cinema.controller;
+
+import com.dev.cinema.model.dto.OrderResponseDto;
+import com.dev.cinema.model.entity.ShoppingCart;
+import com.dev.cinema.model.entity.User;
+import com.dev.cinema.service.interfaces.OrderService;
+import com.dev.cinema.service.interfaces.ShoppingCartService;
+import com.dev.cinema.service.mapper.OrderMapper;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final ShoppingCartService cartService;
+    private final OrderMapper mapper;
+
+    @Autowired
+    public OrderController(OrderService orderService,
+                           ShoppingCartService cartService, OrderMapper mapper) {
+        this.orderService = orderService;
+        this.cartService = cartService;
+        this.mapper = mapper;
+    }
+
+    @PostMapping("/complete")
+    public void completeOrder(@RequestParam Long userId) {
+        ShoppingCart shoppingCart = cartService.getByUser(new User(userId));
+        orderService.completeOrder(shoppingCart);
+    }
+
+    @GetMapping
+    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
+        return orderService.getOrderHistory(new User(userId)).stream()
```

Comment:
```suggestion
        return orderService.getOrderHistory(new User(userId)).stream()
```
I know that you only need userId to find his orders, but let's pass the whole user to this method

Reply:
done